yajsapi/runner/index.ts

```diff
@@ -608,7 +608,8 @@ export class Engine {
         await sleep(2);
         if (
           Object.keys(offer_buffer).length > 0 &&
-          workers.size < self._conf.max_workers
+          workers.size < self._conf.max_workers &&
+          work_queue.has_unassigned_tasks()
```

Comment:
`work_queue.has_unassigned_items()` would be better IMO, since this class uses `items` in several field names (they're not part of the public API but still)

Reply:
Good idea!